cardinal/ecs/engine.go

```diff
@@ -707,6 +707,17 @@ func (e *Engine) Shutdown() {
 	if e.eventHub != nil {
 		e.eventHub.ShutdownEventHub()
 	}
+	log.Info().Msg("Closing storage connections.")
+	errs := errors.Join(
+		e.redisStorage.Close(),
+		e.entityStore.Close(),
```

Comment:
This will result in an error that redis is already closed because we share a client. For now you should only Close one and not the other. 

We can work out the details of this in a later PR. Ideally GameManager and this should go in another PR but if you need this up now, go ahead.

Reply:
i updated it to just close the redisStorage object. also `entityStore.Close` doesn't return an error in the event that it's client was already closed.

```go
// Close closes the manager.
func (m *Manager) Close() error {
	err := eris.Wrap(m.client.Close(), "")
	if eris.Is(eris.Cause(err), redis.ErrClosed) {
		// if redis is already closed that means another shutdown pathway got to it first.
		// There are multiple modules that will try to shutdown redis, if it is already shutdown it is not an error.
		return nil
	}
	return err
}
```